src/v/cluster/partition.h

```diff
@@ -322,7 +294,7 @@ class partition {
     ss::sharded<feature_table>& _feature_table;
     bool _is_tx_enabled{false};
     bool _is_idempotence_enabled{false};
-    ss::lw_shared_ptr<cloud_storage::remote_partition> _cloud_storage_partition;
+    ss::shared_ptr<cloud_storage::remote_partition> _cloud_storage_partition;
```

Comment:
from `shared_ptr.hh`:

> lw_shared_ptr<> is the more lightweight variant, with a lw_shared_ptr<>
> // occupying just one machine word, and adding just one word to the shared
> // object.  However, it does not support polymorphism.
> //
> // shared_ptr<> is more expensive, with a pointer occupying two machine
> // words, and with two words of overhead in the shared object.  In return,
> // it does support polymorphism.

Looks like it will affect Redpanda performance. Do you know how we can test it?

Reply:
I don't think it will make any difference because we don't copy `remote_partition` very often. Compared to other things that this code is doing the difference between `shared_ptr` and `lw_shared_ptr` will be negligible.